### ISSUE TITLE
Python3 unicode support

### ISSUE THREAD

**Issue by verylazyguy** [Vector35/binaryninja-api#1391]
Every API call in the python bindings calls `cstr` when going to C code and `pyNativeStr` when returning. Because these functions use "charmap" encoding, emojis can't be used for anything from the python API (e.g. comments, background progress, `bv.find_next_text`). Can it either be hardcoded to utf8 or have it be a setting that defaults to charmap?

**Comment by KyleMiles** [comment]
So the answer for this is to define pynativestr differently for python2 and python3. Python3 should use utf8, while python2 uses an actually for loop to decode the bytes.

**Comment by KyleMiles** [comment]
I should actually further say that this isn't true... The following snippet generates a `UnicodeDecodeError`:
```
for i in range(256):
   b = i.to_bytes(1, "big")
   b.decode("utf8")
```
So just utf8 isn't the right encoding either.
Anyways, I'll be pushing a fix for this later today.

**Comment by verylazyguy** [comment]
Of course, some bytes indicate that it's a multi-byte character. It's not meant for converting bytes in the binary, it's for displaying user defined things mostly

**Comment by KyleMiles** [comment]
Should be fixed in 1.1.1768-dev.
